### PR TITLE
DAOS-11858 test: Add fault injection test for container get-prop and cp.

### PR DESCRIPTION
**Pull request by ashleypittman**
Add a fault injection test for "daos container get-prop" command
and "cp" under the interception library.

Fix leaks and incorrect logging in the acl handling code.
Resolve pylint warnings in fault injection test code.

### CODE REVIEW

**Review by tanabarr — DISMISSED** [review]

File: src/control/cmd/daos/acl.go [hunk 1]
```diff
@@ -10,12 +10,14 @@ package main
 #include "util.h"
 
 void
-free_strings(char **str, size_t str_count)
+free_strings(char **str, size_t str_count, bool free_array)
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/cmd/daos/acl.go:13]: In cgo, you can put inline functions in [the preamble](https://pkg.go.dev/cmd/cgo#hdr-Using_cgo_with_the_go_command). This is a common convention for little helpers or bridge functions.

[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/control/cmd/daos/acl.go:13]: It'll be Monday now.

**Review by mjmac — APPROVED** [review]

**Comment by ashleypittman** [comment]
Finally got this working properly in Jenkins with the new filenames as well.  The log files are a lot easier to locate/parse now.

File: utils/node_local_test.py [hunk 2]
```diff
@@ -3768,7 +3768,7 @@ def __init__(self, aft, cmd, env, loc):
         # The subprocess handle
         self._sp = None
         # The valgrind handle
-        self.vh = None
+        self.valgrindh = None
```
Comment (daltonbohning) [inline utils/node_local_test.py:3771]: Suggestion - Personally, I prefer self-explanatory names such as `valgrind_handle`, `valgrind_hdl`, etc.

File: utils/node_local_test.py [hunk 3]
```diff
@@ -4006,6 +4008,15 @@ def __init__(self, conf, desc, cmd):
         self.wf = conf.wf
         # Instruct the fault injection code to skip daos_init().
         self.skip_daos_init = True
+        log_dir = f'dnt_fi_{self.description}_logs'
+        if conf.tmp_dir:
+            self.log_dir = join(conf.tmp_dir, log_dir)
+        else:
+            self.log_dir = join('/tmp', log_dir)
```
Comment (daltonbohning) [inline utils/node_local_test.py:4015]: Just FYI that this logic works in python
```
        self.log_dir = join(conf.tmp_dir or '/tmp', log_dir)
```

File: utils/node_local_test.py [hunk 4]
```diff
@@ -4248,6 +4300,28 @@ def test_fi_list_attr(server, conf, wf):
     return rc
 
 
+def test_fi_get_prop(server, conf, wf):
+    """Run daos cont list-attr with fault injection"""
```
Comment (daltonbohning) [inline utils/node_local_test.py:4304]:
```suggestion
    """Run daos cont get-prop with fault injection"""
```

**Review by daltonbohning — APPROVED** [review]
Just aesthetic suggestions

**Pull request merged**